### PR TITLE
Add plist_options and test for logrotate

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/logrotate.rb [hunk 1]
```diff
@@ -63,6 +65,21 @@ def plist; <<-EOS.undent
...
+    # Generate a test log file of size >= 1byte
+    (testpath/"test.log").write("testlograndomstring")
+    # Logrotate config file which rotates the log file of size >= 1 byte
+    (testpath/"testlogrotate.conf").write <<-EOS.undent
+        #{testpath}/test.log {
+        size 1
+        copytruncate
+    }
+    EOS
+    # Rotate the log file and check if the new file's size is 0 bytes
```
Comment (MikeMcQuaid) [inline Formula/logrotate.rb]: Can remove all the comments above, they are good but pretty self-explanatory. Thanks!

File: Formula/logrotate.rb [hunk 2]
```diff
@@ -63,6 +65,21 @@ def plist; <<-EOS.undent
...
+    system "#{sbin}/logrotate", "-s", "#{testpath}/logstatus", "#{testpath}/testlogrotate.conf"
```
Comment (MikeMcQuaid) [inline Formula/logrotate.rb]: Can probably get away without specifying the `#{testpath}/`s here.

File: Formula/logrotate.rb [hunk 3]
```diff
@@ -63,6 +65,21 @@ def plist; <<-EOS.undent
...
+    assert(File.size?("#{testpath}/test.log").nil?)
```
Comment (MikeMcQuaid) [inline Formula/logrotate.rb]: can you add a second argument here that says something like `"File is not zero length!"` so on failure there's a better error message

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Thanks! Looking good so far but I've got a few comments.

File: Formula/logrotate.rb (hunk 3 above)
Comment (GauthamGoli) [inline Formula/logrotate.rb]: I have made the changes accordingly. Thanks for the review!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `logrotate: add plist_options and test`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request merged**